Composer/packages/client/src/recoilModel/dispatchers/provisionQNA.ts

```diff
@@ -0,0 +1,220 @@
+      const startTime = new Date().getTime();
+
+      const notification = createNotification(getPendingQNANotificationCardProps());
+      // add that notification
+      addNotificationInternal(callbackHelpers, notification);
+
+      try {
+        const qnaMakerSearchName = `${resourceName}-search`.toLowerCase().replace('_', '');
+        const qnaMakerWebAppName = `${resourceName}-qnahost`.toLowerCase().replace('_', '');
+        const qnaMakerServiceName = `${resourceName}-qna`;
+
+        const searchManagementClient = new SearchManagementClient(tokenCredentials as any, subscriptionId);
+        await searchManagementClient.services.createOrUpdate(resourceGroupName, qnaMakerSearchName, {
+          location: region,
+          sku: {
+            name: 'standard',
+          },
+          replicaCount: 1,
+          partitionCount: 1,
+          hostingMode: 'default',
+        });
+
+        const webSiteManagementClient = new WebSiteManagementClient(tokenCredentials, subscriptionId);
+        await webSiteManagementClient.appServicePlans.createOrUpdate(resourceGroupName, resourceGroupName, {
+          location: region,
+          sku: {
+            name: 'S1',
+            tier: 'Standard',
+            size: 'S1',
+            family: 'S',
+            capacity: 1,
+          },
+        });
+        // deploy or update exisiting app insights component
+        const applicationInsightsManagementClient = new ApplicationInsightsManagementClient(
+          tokenCredentials,
+          subscriptionId
+        );
+        await applicationInsightsManagementClient.components.createOrUpdate(resourceGroupName, resourceGroupName, {
+          location: region,
+          applicationType: 'web',
+          kind: 'web',
+        });
+
+        // add web config for websites
+        const azureSearchAdminKey = (await searchManagementClient.adminKeys.get(resourceGroupName, qnaMakerSearchName))
+          .primaryKey;
+        const appInsightsComponent = await applicationInsightsManagementClient.components.get(
+          resourceGroupName,
+          resourceGroupName
+        );
+        const userAppInsightsKey = appInsightsComponent.instrumentationKey;
+        const userAppInsightsName = resourceGroupName;
+        const userAppInsightsAppId = appInsightsComponent.appId;
+        const primaryEndpointKey = `${qnaMakerWebAppName}-PrimaryEndpointKey`;
+        const secondaryEndpointKey = `${qnaMakerWebAppName}-SecondaryEndpointKey`;
+        const defaultAnswer = 'No good match found in KB.';
+        const QNAMAKER_EXTENSION_VERSION = 'latest';
+
+        // deploy qna host webapp
+        const webAppResult = await webSiteManagementClient.webApps.createOrUpdate(
+          resourceGroupName,
+          qnaMakerWebAppName,
+          {
+            name: qnaMakerWebAppName,
+            serverFarmId: resourceGroupName,
+            location: region,
+            siteConfig: {
+              cors: {
+                allowedOrigins: ['*'],
```

Comment:
Is this something we want? Setting the CORS policy for a service to \* can be risky. Maybe allow just the MSAppID of the bot.

Reply:
This is how we provision QNA during the provisioning process, and in our ARM templates.